### PR TITLE
[JENKINS-48466] Provide JUnit 5 support for `JenkinsRule`

### PR DESCRIPTION
**Pull request by basil**
See https://github.com/jenkinsci/jenkins-test-harness/pull/438 for the PR description.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6657"></a>

### CODE REVIEW

File: test/pom.xml
Comment (offa) [inline test/pom.xml:195]: [`junit-jupiter` is an aggregator](https://junit.org/junit5/docs/current/user-guide/#dependency-metadata-junit-jupiter) which pulls `junit-jupiter-api` and `junit-jupiter-engine` listed below.

**Review by offa — APPROVED** [review]

File: test/pom.xml
Comment (daniel-beck) [reply test/pom.xml:195]: Why are you adding them separately when using the aggregator?

Comment (basil) [reply test/pom.xml:195]: To be consistent with `cli/pom.xml`, which is code I did not write.

Comment (daniel-beck) [reply test/pom.xml:195]: https://github.com/jenkinsci/jenkins/pull/6842

Comment (basil) [reply test/pom.xml:195]: @daniel-beck In the future, I am asking that you refrain from commenting on my PRs with (implicit) requests for change regarding pre-existing issues.

File: test/pom.xml
Comment (basil) [inline test/pom.xml:203]: Consistent with https://github.com/jenkinsci/jenkins/blob/38d4e86a80d123d50c966f4584a7e795b6bcb227/core/pom.xml#L518-L529=.

File: test/pom.xml
Comment (daniel-beck) [reply test/pom.xml:194]: It wasn't. I misread offa's comment as yours and providing an explanation, rather than as reviewer's feedback. That explanation made no sense to me, hence the comment.

**Review by NotMyFault — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
